lib/extension-core.js

```diff
@@ -96,4 +96,22 @@ function core(loader) {
 
     return Promise.resolve(loaderLocate.call(this, load));
   }
+
+  function applyExtensions(extensions, loader) {
+    loader._extensions = [];
+    for(var i = 0, len = extensions.length; i < len; i++) {
+      extensions[i](loader);
+    }
+  }
+
+  loader._extensions = [core];
```

Comment:
Perhaps we just need that in the `register` and `meta` extensions - ones that are truly transferrable.